_posts/2021-02-10-native-k8s-with-ha.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title:  "How to natively deploy Flink on Kubernetes with HA"
+date:   2021-02-10 00:00:00
+authors:
+- Yang Wang:
+  name: "Yang Wang"
+excerpt: Kubernetes provides built-in functionalities that Flink can leverage for JobManager failover. In Flink 1.12 (FLIP-144), the community implemented a Kubernetes HA service as an alternative to ZooKeeper for highly available production setups. In this blogpost, we will have a close look at how to deploy Flink applications natively on Kubernetes cluster with HA.
+---
+
+Flink has supported resource management systems like YARN and Mesos since the early days; however, these were not designed for the fast-moving cloud-native architectures that are increasingly gaining popularity these days, or the growing need to support complex, mixed workloads (e.g. batch, streaming, deep learning, web services).
+For these reasons, more and more users are turning to Kubernetes to automate the deployment, scaling and management of their Flink applications.
+
+From release to release, the Flink community has made significant progress in integrating natively with Kubernetes, from active resource management to “Zookeeperless” High Availability (HA).
+In this blogpost, we will go through the technical details of deploying Flink applications natively on Kubernetes.
+Then, we’ll deep dive into Flink’s Kubernetes High Availability (HA) architecture and walk through a hands-on example of running a Flink [application cluster](https://ci.apache.org/projects/flink/flink-docs-stable/concepts/glossary.html#flink-application-cluster) on Kubernetes with HA enabled.
```

Comment:
```suggestion
Then, we’ll deep dive into Flink’s Kubernetes High Availability (HA) architecture and walk through a hands-on example of running a Flink [application cluster](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/#application-mode) on Kubernetes with HA enabled.
```

I suggest linking `application cluster` to the deployment docs. We rewrote the Deployment section recently. IMHO, it's a better entrypoint to do further reading.